actor/builtin/miner/miner_test.go

```diff
@@ -490,6 +493,120 @@ func TestMinerSubmitPoStSectorSetUpdates(t *testing.T) {
+	assert.NoError(t, err)
+	assert.NoError(t, res.ExecutionError)
+	assert.Equal(t, uint8(0), res.Receipt.ExitCode)
+	t.Run("power is 3 sectors after third submitPoSt", func(t *testing.T) {
+		expectedPower := types.NewBytesAmount(3).Mul(types.OneKiBSectorSize)
+		actualPower := requireMinerPowerFromChain(t, minerAddr, st, vms, thirdProvingPeriodStart+3, ancestors)
+		assert.Equal(t, expectedPower, actualPower)
+
+		totalStorage := requireTotalStorageFromChain(t, st, vms, thirdProvingPeriodStart+3, ancestors)
+		assert.Equal(t, totalStorageStart.Add(expectedPower), totalStorage)
+
+	})
+
+	// submitPoSt over 1 sector proving set
+	res, err = th.CreateAndApplyTestMessage(t, st, vms, minerAddr, 0, fourthProvingPeriodStart+2, "submitPoSt", ancestors, []types.PoStProof{proof}, types.EmptyIntSet())
+	assert.NoError(t, err)
+	assert.NoError(t, res.ExecutionError)
+	assert.Equal(t, uint8(0), res.Receipt.ExitCode)	
+	
+	t.Run("power is reduced after sectors removed from proving-set", func(t *testing.T) {
+		expectedPower := types.OneKiBSectorSize
+		actualPower := requireMinerPowerFromChain(t, minerAddr, st, vms, fourthProvingPeriodStart+3, ancestors)
+		assert.Equal(t, expectedPower, actualPower)
+
+		totalStorage := requireTotalStorageFromChain(t, st, vms, fourthProvingPeriodStart+3, ancestors)
+		assert.Equal(t, totalStorageStart.Add(expectedPower), totalStorage)
+
+	})
+}
+
+
+func TestMinerSubmitPoStSectorDoneSetUpdates(t *testing.T) {
+	tf.UnitTest(t)
+
+	ctx := context.Background()
+	st, vms := th.RequireCreateStorages(ctx, t)
+
+	ancestors := th.RequireTipSetChain(t, 10)
+	origPid := th.RequireRandomPeerID(t)
+	minerAddr := th.CreateTestMiner(t, st, vms, address.TestAddress, origPid)
+	proof := th.MakeRandomPoStProofForTest()
+	done := types.EmptyIntSet()
+
+	firstCommitBlockHeight := uint64(3)
+	secondProvingPeriodStart := LargestSectorSizeProvingPeriodBlocks + firstCommitBlockHeight
+	thirdProvingPeriodStart := 2*LargestSectorSizeProvingPeriodBlocks + firstCommitBlockHeight
+	fourthProvingPeriodStart := 3*LargestSectorSizeProvingPeriodBlocks + firstCommitBlockHeight
+
+	// add a sector
+	res, err := th.CreateAndApplyTestMessage(t, st, vms, minerAddr, 0, firstCommitBlockHeight, "commitSector", ancestors, uint64(1), th.MakeCommitment(), th.MakeCommitment(), th.MakeCommitment(), th.MakeRandomBytes(types.TwoPoRepProofPartitions.ProofLen()))
+	require.NoError(t, err)
+	require.NoError(t, res.ExecutionError)
+	require.Equal(t, uint8(0), res.Receipt.ExitCode)
+
+	// add another sector
+	res, err = th.CreateAndApplyTestMessage(t, st, vms, minerAddr, 0, firstCommitBlockHeight+1, "commitSector", ancestors, uint64(2), th.MakeCommitment(), th.MakeCommitment(), th.MakeCommitment(), th.MakeRandomBytes(types.TwoPoRepProofPartitions.ProofLen()))
+	require.NoError(t, err)
+	require.NoError(t, res.ExecutionError)
+
+	t.Run("proving-set contains only the first sector committed before submitPoSt", func(t *testing.T) {
```

Comment:
See last comment. If the method described above could also return the proving set from state, you could use it throughout this test as well. There are fewer dependencies between subtests here, but I still don't like them.